packages/shared/profiles/sagas.ts

```diff
@@ -183,33 +183,67 @@ export function* handleFetchProfile(action: ProfileRequestAction): any {
   }
 }
 
-function* getRemoteProfile(userId: string, version?: number) {
+function* getRemoteProfile(
+  userId: string,
+  version?: number
+): Generator<CallEffect<RemoteProfile> | CallEffect<Array<Avatar>> | RemoteProfile> {
   try {
-    const profiles: { avatars: Avatar[] } = yield call(profileServerRequest, userId, version)
+    const remoteProfile: RemoteProfile = (yield call(profileServerRequest, userId, version)) as RemoteProfile
 
-    let avatar = profiles.avatars[0]
+    const profiles: Avatar[] = (yield call(processRemoteProfiles, [remoteProfile], [userId])) as Array<Avatar>
+    return profiles[0]
+  } catch (error: any) {
+    if (error.message !== 'Profiles not found') {
+      defaultLogger.log(`Error requesting profile for auth check ${userId}, `, error)
+    }
+  }
+  return null
+}
 
-    if (avatar) {
-      avatar = ensureAvatarCompatibilityFormat(avatar)
-      if (!validateAvatar(avatar)) {
-        defaultLogger.warn(`Remote avatar for user is invalid.`, userId, avatar, validateAvatar.errors)
+async function processRemoteProfiles(profiles: RemoteProfile[], userIds: string[]): Promise<Array<Avatar>> {
+  const avatars: Array<Avatar> = profiles
+    .map((profile): Avatar | null => {
+      let avatar = profile.avatars[0]
+      if (!avatar) {
+        return null
+      }
+
+      try {
+        avatar = ensureAvatarCompatibilityFormat(avatar)
+        if (!validateAvatar(avatar)) {
+          defaultLogger.warn(`Remote avatar for users is invalid.`, avatar, validateAvatar.errors)
+          trackEvent(REMOTE_AVATAR_IS_INVALID, {
+            avatar
+          })
+          return null
+        }
+      } catch (error) {
+        defaultLogger.warn(`Remote avatar for users is invalid.`, avatar, validateAvatar.errors)
+        trackEvent(REMOTE_AVATAR_IS_INVALID, {
+          avatar
+        })
         return null
       }
 
       avatar.hasClaimedName = !!avatar.name && avatar.hasClaimedName // old lambdas profiles don't have claimed names if they don't have the "name" property
       avatar.hasConnectedWeb3 = true
 
       return avatar
-    }
-  } catch (error: any) {
-    if (error.message !== 'Profile not found') {
-      defaultLogger.log(`Error requesting profile for auth check ${userId}, `, error)
-    }
-  }
-  return null
+    })
+    .filter((avatar: Avatar | null): boolean => avatar !== null) as Array<Avatar>
+
+  // generate a random profile for users that don't have one
+  // this is so that renderer can show them to the user
+  // and show them in the chat so no ghost notification is left hanging
+  const usersWithoutAvatars = userIds.filter((userId) => !avatars.some((avatar) => avatar.userId === userId))
+  const defaultAvatars = await Promise.all(
+    usersWithoutAvatars.map(async (userId): Promise<Avatar> => generateRandomUserProfile(userId))
+  )
```

Comment:
I think this should not be part of the logic of `processRemoteProfiles` because it alters the initial loading of the "self" profile

Reply:
Sorry, this is a leftover from a rollbacked change. Will remove it now.